tests/serverpod_test_server/test/generated_methods/from_json_test.dart

```diff
@@ -0,0 +1,289 @@
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:test/test.dart';
+
+void main() {
```

Comment:
Discussion: It is hard to understand if we cover all the scenarios required in the tests, perhaps we could use `Group` and create familiar patterns which makes it easy to understand what we cover.

We preferebly want one deserialization test for each types we support, and the tests should cover all paths (correct value, incorrect value, already parsed value as is possible for DateTime for example). If we could create a group for each type it makes it a bit more clear what is tested.

The types that I can think of that need testing are:
- Int
- Double
- String
- Bool
- DateTime
- ByteData
- Duration
- UuidValue
- Enum (Serialized as int)
- Enum (Serialized as String)
- Class
- Map
- List

And a possible group could be `Given int field`, then a test could be called `when deserializing correctly formatted int value then field contains value.`

Reply:
added more tests but i didn't group these tests, let me know if it is still not okay?!